wf-galleon/WFLY-19021-Stability_In_Provisioning.adoc

```diff
@@ -0,0 +1,194 @@
+* To filter-out packages at a stability level lower than the minimum-stability level. Such packages are not packaged inside the feature-pack.
+
+==== Stability for WildFly feature-packs
+
+Expected minimum stability levels for WildFly feature-packs:
+
+* wildfly-ee-galleon-pack: `experimental`
+* wildfly-galleon-pack: `experimental`
+* wildfly-preview-feature-pack: `experimental`
+
+=== Requirements at provisioning time
+
+Provisioning time follows some rules to properly handle stability and avoid features/packages of wrong stability being provisioned.
+
+* The feature-pack contained minimum stability level is used to constrain the set of provisioned features. Any features/packages at a lower stability level contained in the 
+feature-pack (if any), are not provisioned.
+* By default, the minimum stability level of each feature-pack is used when provisioning its own content. 
+* Each feature-pack present in the provisioning configuration can have a different minimum stability level. 
+The minimum stability of each feature-pack applies to its own content and doesn't leak into other feature-packs.
+* A user can specify a stability-level, by mean of the `stability-level` Galleon option. 
+This stability level is used to constrain the stability level of all provisioned features/packages. If the specified level is lower than a feature-pack minimum stability, 
+the feature-pack minimum stability is used.
+
+For example, a provisioning configuration containing 3 feature-packs:
+
+* A, minimum stability default
+* B, minimum stability community
+* C, minimum stability experimental
+
+If no stability level is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+If the stability level `default` is specified:
+
+* `default` features/packages of A are provisioned
+* `default` features/packages of B are provisioned
+* `default` features/packages of C are provisioned
+
+If the stability level `experimental` is specified:
+
+* `default` features/packages of A are provisioned
+* `community` and `default` features/packages of B are provisioned
+* `experimental`, `preview`, `community` and `default` features/packages of C are provisioned
+
+==== Lowest stability level needed by WildFly Galleon Plugins
+
+At provisioning time, when WildFly Galleon plugin generates the server configurations, 
+a WildFly embedded server is started. This server is started with a stability level computed in the following way:
+
+* If a stability level has been specified by the user, this level is used.
+* If no stability level has been specified, the lowest minimum stability level of all feature-packs present in the provisioning is used.
+
+=== Provisioning tooling requirement
+
+* All Galleon based tooling should let users specify the `stability-level` option.
```

Comment:
@jfdenise @jamezp @spyrkob is this achievable by March 20, in time for the WF 32 feature freeze?

If it is not, or is doubtful, for some, but not all, tooling we should enumerate the "must have" tooling here and list others in "Nice-to-Have" or "Non-Requirements". If there are Non-Requirements (or later we realize some Nice-to-Have won't happen) then I like the practice of adding a "Future Work" section to give a hint at what may be coming later.

Reply:
I listed the tools I know that will be ok for WF32. These tools already switched to Galleon 6. So the support for stability is a mater of bumping the Galleon version to the latest 6 release (that also supports upward compat).
Prospero is still using Galleon 5.x, so we should port the feature to Galleoin 5.x or have prospero to move to Galleon 6 (There is a wip branch for it).